test/playlist.test.js

```diff
@@ -686,6 +686,95 @@ QUnit.test('estimates segment request time based on bandwidth', function(assert)
   assert.equal(estimate, 8, 'takes into account bytes already received from download');
 });
 
+QUnit.module('Playlist enabled states', {
+  beforeEach(assert) {
+    this.env = useFakeEnvironment(assert);
+    this.clock = this.env.clock;
+  },
+  afterEach() {
+    this.env.restore();
+  }
+});
+
+QUnit.test('determines if a playlist is incompatible', function(assert) {
+  // incompatible means that the playlist was blacklisted due to incompatible
+  // configuration e.g. audio only stream when trying to playback audio and video.
+  // incompaatibility is denoted by a blacklist of Infinity.
+  const playlist = {};
+
+  assert.notOk(Playlist.isIncompatible(playlist),
+    'playlist not incompatible if no excludeUntil');
+
+  playlist.excludeUntil = 1;
+  assert.notOk(Playlist.isIncompatible(playlist),
+    'playlist not incompatible if expired blacklist');
+
+  playlist.excludeUntil = Date.now() + 9999;
+  assert.notOk(Playlist.isIncompatible(playlist),
+    'playlist not incompatible if temporarily blacklisted');
+
+  playlist.excludeUntil = Infinity;
+  assert.ok(Playlist.isIncompatible(playlist),
+    'playlist is incompatible if excludeUntil is Infinity');
+});
+
+QUnit.test('determines if a playlist is blacklisted', function(assert) {
+  const playlist = {};
+
+  assert.notOk(Playlist.isBlacklisted(playlist),
+    'playlist not blacklisted if no excludeUntil');
+
+  playlist.excludeUntil = Date.now() - 1;
+  assert.notOk(Playlist.isBlacklisted(playlist),
+    'playlist not blacklisted if expired excludeUntil');
+
+  playlist.excludeUntil = Date.now() + 9999;
+  assert.ok(Playlist.isBlacklisted(playlist),
+    'playlist is blacklisted');
+
+  playlist.excludeUntil = Infinity;
+  assert.ok(Playlist.isBlacklisted(playlist),
+    'playlist is blacklisted if excludeUntil is Infinity');
+});
+
+QUnit.test('determines if a playlist is disabled', function(assert) {
+  const playlist = {};
+
+  assert.notOk(Playlist.isDisabled(playlist), 'playlist not disabled');
+
+  playlist.disabled = true;
+  assert.ok(Playlist.isDisabled(playlist), 'playlist is disabled');
+});
+
+QUnit.test('determines if a playlist is enabled', function(assert) {
+  // enabled means not blacklisted and not disabled
+  const playlist = {};
+
+  assert.ok(Playlist.isEnabled(playlist), 'playlist is enabled');
+
+  playlist.excludeUntil = Date.now() - 1;
+  assert.ok(Playlist.isEnabled(playlist), 'playlist is enabled');
+
+  playlist.excludeUntil = Date.now() + 9999;
+  assert.notOk(Playlist.isEnabled(playlist), 'playlist is not enabled');
+
+  playlist.excludeUntil = Infinity;
+  assert.notOk(Playlist.isEnabled(playlist), 'playlist is not enabled');
+
+  delete playlist.excludeUntil;
```

Comment:
Could we maybe break this test up in some way?  Maybe one for if playerlist.disabled is set to true? Or maybe based on the expectation (should be enabled vs should not be enabled)?

It also might be safer to not maintain the `const playlist` here and instead pass the raw values for `playlist` in.  As an example:

```
playlist.excludeUntil = Date.now() - 1;
assert.ok(Playlist.isEnabled(playlist), 'playlist is enabled');
```

Just becomes:
```
assert.ok(Playlist.isEnabled({ excludeUntil: Date.now() - 1 }), 'playlist is enabled');
```

This way you don't need to worry about deleting things and the order of the tests won't matter.